### DISCUSSION TITLE
Not working on private GitHub repos

### DISCUSSION THREAD

**Discussion by Ayfri** [badges/shields#5676] [Q&A] [answered]
Are you experiencing an issue with...

- [x] [shields.io](https://shields.io/#/)
- [ ] My own instance
- [ ] [badge-maker NPM package](https://www.npmjs.com/package/badge-maker)

:beetle: **Description**
Hi!
I tried to add a badge with this website for a private repo on an organization that I have but on the website, it always says that it cannot found the repo (I didn't try on GitHub).

:link: https://img.shields.io/github/repo-size/Galileo-Bot/galileo
I think it is useless to add the badge as you can't see the repo but I don't know.

:bulb: I don't know :/

**Comment by calebcartwright** [comment, marked as answer]
hi @Ayfri :wave: The Shields.io service can only provide badges for publicly accessible targets (repos, projects, builds, etc.), as the Shields.io server doesn't have access to your private GitHub data either. 

In order to get badges for your private content, you'd need to deploy your own self-hosted Shields server that's configured with your own GitHub Authentication configuration. More details on how to do that can be found [here](https://github.com/badges/shields/blob/master/doc/self-hosting.md)

**Reply by dmtzs** [reply]
I have the same doubt but in you answer you only redirect to a way to host your own badges but not about how to authenticate with github, is there a doc or something about how to do that? Or I'm misunderstanding something?

**Reply by calebcartwright** [reply]
> I have the same doubt but in you answer you only redirect to a way to host your own badges but not about how to authenticate with github, is there a doc or something about how to do that? Or I'm misunderstanding something?

I'm confused by your question. If you read the linked document, you'll see see a Server Secrets section https://github.com/badges/shields/blob/master/doc/self-hosting.md#server-secrets. That document describes the process of configuring a shields server, with a specific section for the GitHub details in https://github.com/badges/shields/blob/master/doc/server-secrets.md#github
